lib/index.js

```diff
@@ -73,50 +73,68 @@ function highlight (sqlString, options) {
       matches.push({
         name: hl.name,
         start: match.index + boringLength,
-        length: (hl.trimEnd ? text.substr(0, text.length - hl.trimEnd) : text).length
+        length: (hl.trimEnd
+          ? text.substring(0, text.length - hl.trimEnd)
+          : text
+        ).length
       })
     }
   }
 
   const sortedMatches = matches.slice().sort((a, b) => a.start - b.start)
 
   // filter/exclude nested matches (matches within the last match)
-  const filteredMatches = []
+  const segments = []
   let upperBound = 0
   for (let i = 0; i < sortedMatches.length; i++) {
     if (sortedMatches[i].start >= upperBound) {
-      filteredMatches.push(sortedMatches[i])
+      if (sortedMatches[i].start > upperBound) {
+        segments.push({
+          name: DEFAULT_KEYWORD,
+          content: sqlString.substring(upperBound, sortedMatches[i].start)
+        })
+      }
+
+      segments.push({
+        name: sortedMatches[i].name,
+        content: sqlString.substring(
+          sortedMatches[i].start,
+          sortedMatches[i].start + sortedMatches[i].length
+        )
+      })
       upperBound = sortedMatches[i].start + sortedMatches[i].length
     }
   }
```

Comment:
To avoid nesting if statements, perhaps we could break if there is nothing more to process?

```js
for (let i = 0; i < sortedMatches.length; i++) {
  if (sortedMatches[i].start < upperBound) { break }

  // If no match, add a whitespace segment
  if (sortedMatches[i].start > upperBound) {
    segments.push({
      name: DEFAULT_KEYWORD,
      content: sqlString.substring(upperBound, sortedMatches[i].start)
    })
  }

  segments.push({
    name: sortedMatches[i].name,
    content: sqlString.substring(
      sortedMatches[i].start,
      sortedMatches[i].start + sortedMatches[i].length
    )
  })
  upperBound = sortedMatches[i].start + sortedMatches[i].length
}
```